CRM/Core/BAO/Address.php

```diff
@@ -362,127 +362,88 @@ public static function &getValues($entityBlock, $microformat = FALSE, $fieldName
-      }
-
-      if (!empty($address->country_id)) {
-        $address->country = CRM_Core_PseudoConstant::country($address->country_id);
-        $values['country'] = $address->country;
-
-        //get world region
-        $regionId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Country', $address->country_id, 'region_id');
-        $values['world_region'] = CRM_Core_PseudoConstant::worldregion($regionId);
-      }
+      // add state and country information: CRM-369. Also necessary for vCard format.
+      // FIXME: Longer-term, CRM_Utils_Address::format and formatVCard should use API4 field names.
+      $values['location_type'] = $address['location_type_id:label'];
+      $values['state_province_abbreviation'] = $address['state_province_id:abbr'];
+      $values['state_province'] = $values['state_province_name'] = $address['state_province_id:label'];
+      $values['country'] = $address['country_id:label'];
+      $values['world_region'] = $address['country_id.region_id:label'];
 
-      $address->addDisplay($microformat);
+      self::addDisplay($values, $microformat);
 
-      $values['display'] = $address->display;
-      $values['display_text'] = $address->display_text;
-
-      if (isset($address->master_id) && !CRM_Utils_System::isNull($address->master_id)) {
+      if ($address['master_id'] ?? FALSE) {
         $values['use_shared_address'] = 1;
       }
 
-      $addresses[$count] = $values;
-
-      //There should never be more than one primary blocks, hence set is_primary = 0 other than first
-      // Calling functions expect the key is_primary to be set, so do not unset it here!
-      if ($count > 1) {
-        $addresses[$count]['is_primary'] = 0;
-      }
-
+      $formattedAddresses[$count] = $values;
       $count++;
     }
 
-    return $addresses;
+    return $formattedAddresses;
   }
 
   /**
    * Add the formatted address to $this-> display.
    *
+   * @param array $address
+   *
    * @param bool $microformat
    *   Unexplained parameter that I've always wondered about.
    */
-  public function addDisplay($microformat = FALSE) {
-    $fields = [
-      // added this for CRM 1200
-      'address_id' => $this->id,
-      // CRM-4003
-      'address_name' => str_replace('', ' ', ($this->name ?? '')),
-      'street_address' => $this->street_address,
-      'supplemental_address_1' => $this->supplemental_address_1,
-      'supplemental_address_2' => $this->supplemental_address_2,
-      'supplemental_address_3' => $this->supplemental_address_3,
-      'city' => $this->city,
-      'state_province_name' => $this->state_name ?? "",
-      'state_province' => $this->state ?? "",
-      'postal_code' => $this->postal_code ?? "",
-      'postal_code_suffix' => $this->postal_code_suffix ?? "",
-      'country' => $this->country ?? "",
-      'world_region' => $this->world_region ?? "",
-    ];
+  private static function addDisplay(array &$address, bool $microformat = FALSE) {
```

Comment:
Just noting this has gone from public to private - I'm trying to load up universe to look for callers but it needs to index ....  a safer option would be to deprecate this & add a new private function